kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/defaultimpl/repository/impl/KeycloakImpl.java

```diff
@@ -690,4 +691,42 @@ private String getRoleId(String roleName,String realmId) throws IOException {
 		
 	}
 
+	@Override
+	public VidDto getVidFromUserId(String userId, String realmID) {
+		VidDto vIDDto = new VidDto();
+		Map<String, String> pathParams = new HashMap<>();
+		pathParams.put(AuthConstant.REALM_ID, realmID);
+		HttpHeaders httpHeaders = new HttpHeaders();
+		HttpEntity<String> httpEntity = new HttpEntity<>(null, httpHeaders);
+		UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder
+				.fromUriString(keycloakAdminUrl + users + "?username=" + userId);
+		String response = callKeycloakService(uriComponentsBuilder.buildAndExpand(pathParams).toString(),
+				HttpMethod.GET, httpEntity);
+		if (response == null || response.isEmpty()) {
+			throw new AuthManagerException(AuthErrorCode.USER_NOT_FOUND.getErrorCode(),
+					AuthErrorCode.USER_NOT_FOUND.getErrorMessage());
+		}
+		try {
+			JsonNode node = objectMapper.readTree(response);
+			for (JsonNode jsonNode : node) {
+				if (jsonNode.get(AuthConstant.USER_NAME).textValue().equals(userId)) {
+					JsonNode attriNode = jsonNode.get("attributes");
+					String vid = attriNode.get(AuthConstant.VID).get(0).textValue();
+					vIDDto.setVid(vid);
+ 					break;
+				}
+			}
+			if (vIDDto.getVid() == null) {
+				throw new AuthManagerException(AuthErrorCode.USER_NOT_FOUND.getErrorCode(),
```

Comment:
@urviljoshi If the user is found but the RID attribute is not found, can we return a different message like 'RID attribute is not available for the user' ? Returning the 'user not found' in such case will be misleading.

Reply:
sure